frontend/src/metabase/list-view/components/ListView/ListViewConfiguration.tsx

```diff
@@ -406,3 +413,32 @@ export const ListViewConfiguration = ({
     </DndContext>
   );
 };
+
+function generatePreviewSample(rows: RowValues[], cols: Lib.ColumnMetadata[]) {
+  if (!rows.length) {
+    return null;
+  }
+
+  // Start with the first row as our sample
+  const sample = [...rows[0]];
+
+  // Track which positions still need non-null values
+  const nullPositions = new Set<number>();
+  sample.forEach((value, index) => {
+    if (value === null || value === undefined) {
+      nullPositions.add(index);
+    }
+  });
+
+  // If first row has all non-null values, we're done
+  if (nullPositions.size === 0) {
+    return sample;
+  }
+
+  for (const position of nullPositions) {
+    const col = cols[position];
+    sample[position] = getColumnExample(col);
+  }
+
+  return sample;
```

Comment:
```suggestion
  return rows[0].map((value, index) => value == null ? getColumnExample(cols[index]) : value);
```

<img width="1188" height="1134" alt="image" src="https://github.com/user-attachments/assets/e5c65f8b-cf8f-4a36-8ce9-b10a2a3f4064" />